### DISCUSSION TITLE
"accountsChanged" event does not fire in web3modal v3

### DISCUSSION THREAD

**Discussion by uninstalls** [WalletConnect/walletconnect-monorepo#3982] [Developer Support] [answered]
This is v2 solution: https://github.com/WalletConnect/web3modal/issues/405
How to fix in v3? THX

**Comment by glitch-txs** [comment, marked as answer]
You can subscribe to account changes https://docs.walletconnect.com/web3modal/javascript/actions?platform=ethers#subscribe-to-theme-changes

```ts
function handleChange({ address }){
  console.log(address)
}

modal.subscribeProvider(handleChange)
```

**Reply by uninstalls** [reply]
It works!! Thank you very much!! 👍👍👍
